chronos/asyncsync.nim

```diff
@@ -125,8 +118,8 @@ proc release*(lock: AsyncLock) =
   ## other coroutines are blocked waiting for the lock to become unlocked,
   ## allow exactly one of them to proceed.
   if lock.locked:
-    lock.locked = false
-    lock.wakeUpFirst()
+    if not(lock.wakeUpFirst()):
```

Comment:
I think add a comment here explaining the logic why we don't set locked to false - in the new flow, `locked = true` will still run in `acquire` which on a casual reading of the code looks "unbalanced"